monai/networks/utils.py

```diff
@@ -748,6 +748,70 @@ def convert_to_torchscript(
     return script_module
 
 
+def _onnx_trt_compile(
+    onnx_model,
+    min_shape: Sequence[int],
+    opt_shape: Sequence[int],
+    max_shape: Sequence[int],
+    device: int,
+    precision: str,
+    input_names: Sequence[str] | None,
+    output_names: Sequence[str] | None,
+):
+    """
+    onnx_model: the source ONNX model to compile.
+    min_shape: the minimum input shape of the converted TensorRT model.
+    opt_shape: the optimization input shape of the model, on which the TensorRT optimizes.
+    max_shape: the maximum input shape of the converted TensorRT model.
+    device: the target GPU index to convert and verify the model.
+    precision: the weight precision of the converted TensorRT engine-based TorchScript model.
+        Should be 'fp32' or 'fp16'.
+    input_names: optional input names of the ONNX model.
+    output_names: optional output names of the ONNX model.
+    """
+    trt, _ = optional_import("tensorrt", "8.5.3")
+    torch_tensorrt, _ = optional_import("torch_tensorrt", "1.4.0")
+
+    input_shapes = (min_shape, opt_shape, max_shape)
+    # default to an empty list to fit the `torch_tensorrt.ts.embed_engine_in_new_module` function.
+    input_names = [] if not input_names else input_names
+    output_names = [] if not output_names else output_names
+
+    # set up the TensorRT builder
+    torch_tensorrt.set_device(device)
+    logger = trt.Logger(trt.Logger.WARNING)
```

Comment:
Hi @Nic-Ma ,
     The logger is a `tensorrt.Logger` to record the internal state of tensorrt conversion, which is necessary for the `tensorrt.Builder`, instead of a normal python logger. This logic follows the pulic released [API example](https://docs.nvidia.com/deeplearning/tensorrt/developer-guide/index.html#importing_trt_python). The default  log level is the `trt.Logger.WARNING` as shown in the [API doc](https://docs.nvidia.com/deeplearning/tensorrt/api/python_api/infer/Core/Logger.html). And I think it doesn't need to be set back to the original level, since it only works inside the tensorrt conversion and will be released after the conversion.

Thanks,
Bin